docs/examples/spectrum/average_photon_energy.py

```diff
@@ -0,0 +1,198 @@
+"""
+Average Photon Energy Calculation
+============================
+
+Calculation of the Average Photon Energy from SPECTRL2 output.
+"""
+
+# %%
+# Introduction
+# ------------
+# This example demonstrates how to use the
+# :py:func:`~pvlib.spectrum.average_photon_energy` function to calculate the
+# Average Photon Energy (APE, :math:`\overline{E_\gamma}`) of spectral
+# irradiance distributions simulated using :py:func:`~pvlib.spectrum.spectrl2`.
+# More information on the SPECTRL2 model can be found in [1]_.
+# The APE parameter is a useful indicator of the overall shape of the solar
+# spectrum [2]_. Higher (lower) APE values indicate a blue (red) shift in the
+# spectrum and is one of a variety of such characterisation methods that is
+# used in the PV performance literature [3]_.
+#
+# To demonstrate this functionality, first we need to simulate some spectra
+# using :py:func:`~pvlib.spectrum.spectrl2`. In this example, we will simulate
+# spectra following a similar method to that which is followed in the
+# `Modelling Spectral Irradiance
+# <https://pvlib-python.readthedocs.io/en/stable/gallery/spectrum/plot_spectrl2_fig51A.html>`_
```

Comment:
@echedey-ls did you ever figure out how to link to gallery examples via sphinx references?  It would be nice to not hardcode this URL (in part because it will inevitably need updating, and in part because it specifies `stable` no matter how the docs are built).

Reply:
Yes, sir! 🫡 

Relevant documentation: https://sphinx-gallery.github.io/stable/advanced.html#cross-referencing

Manual steps to follow when you want to double check:
1. Run the sphinx html builder as usual
   - you can cancel mid-generation after "Generating gallery for..." messages have finished. They mean `.rst` files for each gallery example have been created
3. Go to `docs/sphinx/source/gallery/` and find your rST file of the example
   - in this case, `.../spectrum/average_photon_energy.rst`
3. Open file and see line 18:
   
   ![image](https://github.com/user-attachments/assets/e9b55704-9fd7-48b9-a7df-546a9cf6f3ea)

4. That's known as a cross-reference; https://bashtage.github.io/sphinx-material/rst-cheatsheet/rst-cheatsheet.html#example
5. Remove first underscore, that's your reference identifier.
6. Enjoy cross-referencing with ``` :ref:`sphx_glr_gallery_spectrum_average_photon_energy.py` ```!

Reply:
Thanks @echedey-ls !